cpp/src/arrow/sparse_tensor_test.cc

```diff
@@ -712,4 +712,194 @@ INSTANTIATE_TYPED_TEST_CASE_P(TestInt64, TestSparseCSRMatrixForIndexValueType, I
+  }
+
+  template <typename CValueType>
+  std::shared_ptr<SparseCSFTensor> MakeSparseTensor(
+      const std::shared_ptr<SparseCSFIndex>& si,
+      std::vector<CValueType>& sparse_values) const {
+    auto data = Buffer::Wrap(sparse_values);
+    return std::make_shared<SparseCSFTensor>(si,
+                                             CTypeTraits<CValueType>::type_singleton(),
+                                             data, this->shape_, this->dim_names_);
+  }
+};
+
+TYPED_TEST_CASE_P(TestSparseCSFTensorForIndexValueType);
+
+TYPED_TEST_P(TestSparseCSFTensorForIndexValueType, TestSparseTensorFromTensor) {
+  using IndexValueType = TypeParam;
+  using c_index_value_type = typename IndexValueType::c_type;
+
+  std::vector<int64_t> data_values = {1, 2, 3, 4, 5, 6, 7, 8};
+  std::vector<std::vector<c_index_value_type>> indptr_values = {
+      {0, 2, 3}, {0, 1, 3, 4}, {0, 2, 4, 5, 8}};
+  std::vector<std::vector<c_index_value_type>> indices_values = {
+      {1, 2}, {1, 2, 2}, {1, 1, 2, 2}, {2, 3, 1, 3, 1, 1, 2, 3}};
+  std::vector<std::shared_ptr<Buffer>> indptr_buffers(3);
+  std::vector<std::shared_ptr<Buffer>> indices_buffers(4);
+  std::vector<int64_t> axis_order = {0, 1, 2, 3};
+  std::vector<int64_t> sparse_tensor_shape({3, 3, 3, 4});
+  std::vector<int64_t> indices_shapes({2, 3, 4, 8});
+  std::vector<std::string> dim_names({"a", "b", "c", "d"});
+
+  std::shared_ptr<Buffer> data_buffer = Buffer::Wrap(data_values);
+  for (int64_t i = 0; i < static_cast<int64_t>(indptr_values.size()); ++i)
+    indptr_buffers[i] = Buffer::Wrap(indptr_values[i]);
+  for (int64_t i = 0; i < static_cast<int64_t>(indices_values.size()); ++i)
+    indices_buffers[i] = Buffer::Wrap(indices_values[i]);
+
+  std::shared_ptr<SparseCSFIndex> sparse_index;
+  ASSERT_OK_AND_ASSIGN(
+      sparse_index,
+      SparseCSFIndex::Make(TypeTraits<IndexValueType>::type_singleton(), indices_shapes,
+                           axis_order, indptr_buffers, indices_buffers));
+  std::shared_ptr<SparseCSFTensor> sparse_tensor = std::make_shared<SparseCSFTensor>(
+      sparse_index, int64(), data_buffer, sparse_tensor_shape, dim_names);
+
+  ASSERT_TRUE(sparse_tensor->Equals(*this->sparse_tensor_from_dense_));
+}
+
+TYPED_TEST_P(TestSparseCSFTensorForIndexValueType, TestSparseTensorToTensor) {
+  using IndexValueType = TypeParam;
+  using c_index_value_type = typename IndexValueType::c_type;
+
+  std::vector<int64_t> data_values = {1, 2, 3, 4, 5, 6, 7, 8};
+  std::vector<std::vector<c_index_value_type>> indptr_values = {
+      {0, 2, 3}, {0, 1, 3, 4}, {0, 2, 4, 5, 8}};
+  std::vector<std::vector<c_index_value_type>> indices_values = {
+      {1, 2}, {1, 2, 2}, {1, 1, 2, 2}, {2, 3, 1, 3, 1, 1, 2, 3}};
+  std::vector<std::shared_ptr<Buffer>> indptr_buffers(3);
+  std::vector<std::shared_ptr<Buffer>> indices_buffers(4);
+  std::vector<int64_t> axis_order = {0, 1, 2, 3};
+  std::vector<int64_t> sparse_tensor_shape({3, 3, 3, 4});
```

Comment:
This shape doesn't involve to change `axis_order`, so we also need to test another shape like `{4, 3, 5, 2}` to test the `axis_order`-changing case.

Reply:
Good idea, will work on this.